### PR TITLE
Refactor hmi message adapter

### PR DESCRIPTION
**Pull request by Kozoriz**
Extracted interface from HMIMessageAdapter
Created HMIMessageAdapterImpl class
Provided tests for HMIMessageAdapterImpl

Related task: APPLINK-20266

Reviewed in : https://github.com/CustomSDL/sdl_panasonic/pull/145

### CODE REVIEW

**Comment by Kozoriz** [comment]
please review: @malirod, @VVeremjova , @OHerasym

File: src/components/hmi_message_handler/src/dbus_message_adapter.cc [hunk 1]
```diff
@@ -33,7 +33,7 @@
 #include "hmi_message_handler/dbus_message_adapter.h"
 #include <sstream>
 #include "utils/logger.h"
-#include "formatters/CSmartFactory.hpp"
+#include "formatters/CSmartFactory.h"
```
Comment (VVeremjova) [inline src/components/hmi_message_handler/src/dbus_message_adapter.cc]: As I see, CSmartFactory is .hpp, not .h. You should rename it

Comment (Kozoriz) [inline src/components/hmi_message_handler/src/dbus_message_adapter.cc]: @VVeremjova , 7875fe9 and fc19ff0

**Comment by malirod** [comment]
Reviewed

**Comment by OHerasym** [comment]
Reviewed

**Comment by VVeremjova** [comment]
Reviewed

**Pull request merged**